iptstate.cc

```diff
@@ -1091,24 +1093,46 @@ int conntrack_hook(enum nf_conntrack_msg_type nf_type, struct nf_conntrack *ct,
     return NFCT_CB_CONTINUE;
   }
 
-  if (flags->filter_src && (memcmp(&(entry->src), &(filters->src), entrysize))) {
-    counts->skipped++;
-    return NFCT_CB_CONTINUE;
-  }
+  if (flags->filter_inv) {
+    if (flags->filter_src && ! (memcmp(&(entry->src), &(filters->src), entrysize))) {
+      counts->skipped++;
+      return NFCT_CB_CONTINUE;
+    }
 
-  if (flags->filter_srcpt && (entry->srcpt != filters->srcpt)) {
-    counts->skipped++;
-    return NFCT_CB_CONTINUE;
-  }
+    if (flags->filter_srcpt && (entry->srcpt == filters->srcpt)) {
```

Comment:
Rather than repeating all of the logic twice, I think something like this would be cleaner:

```
if (flags->filter_srcpt) {
  if (flags->filter_inv && entry->srcpt == filters->srcpt) || (!flags->filter_inv && entry->srcpt != filters->srcpt)
     counts->skipped++;
     return NFCT_CB_CONTINUE;
  }
}
```

Reply:
Perfect, much cleaner indeed.